program/src/processor.rs

```diff
@@ -5987,6 +5987,174 @@ impl Processor {
             referrer_id,
         )
     }
+
+    #[inline(never)]
+    fn cancel_all_spot_orders(program_id: &Pubkey, accounts: &[AccountInfo]) -> MangoResult {
+        const NUM_FIXED: usize = 21;
+
+        let [
+            mango_group_ai, // read
+            mango_cache_ai,   // read
+            mango_account_ai, // write
+            owner_ai, // read, signer
+            base_root_bank_ai,      // read
+            base_node_bank_ai,      // write
+            base_vault_ai,          // write
+            quote_root_bank_ai,     // read
+            quote_node_bank_ai,     // write
+            quote_vault_ai,         // write
+
+            spot_market_ai,         // write
+            bids_ai,                // write
+            asks_ai,                // write
+            open_orders_ai,         // write
+            signer_ai,              // read
+            dex_event_queue_ai,     // write
+            dex_base_ai,            // write
+            dex_quote_ai,           // write
+            dex_signer_ai,          // read
+            dex_prog_ai,            // read
+            token_prog_ai,          // read
+        ] = array_ref![accounts, 0, NUM_FIXED];
+
+        let mango_group = MangoGroup::load_checked(mango_group_ai, program_id)?;
+        check_eq!(dex_prog_ai.key, &mango_group.dex_program_id, MangoErrorCode::InvalidProgramId)?;
+        check!(signer_ai.key == &mango_group.signer_key, MangoErrorCode::InvalidSignerKey)?;
+
+        let mut mango_account =
+            MangoAccount::load_mut_checked(mango_account_ai, program_id, mango_group_ai.key)?;
+        check!(
+            &mango_account.owner == owner_ai.key || &mango_account.delegate == owner_ai.key,
+            MangoErrorCode::InvalidOwner
+        )?;
+        check!(owner_ai.is_signer, MangoErrorCode::InvalidSignerKey)?;
+
+        let market_index = mango_group.find_spot_market_index(spot_market_ai.key).unwrap();
+        check_eq!(
+            &mango_account.spot_open_orders[market_index],
+            open_orders_ai.key,
+            MangoErrorCode::InvalidOpenOrdersAccount
+        )?;
+
+        let signer_seeds = gen_signer_seeds(&mango_group.signer_nonce, mango_group_ai.key);
+        invoke_cancel_orders(
+            open_orders_ai,
+            dex_prog_ai,
+            spot_market_ai,
+            bids_ai,
+            asks_ai,
+            signer_ai,
+            dex_event_queue_ai,
+            &[&signer_seeds],
+            u8::MAX,
```

Comment:
need to pass a `limit: u8`: we may need more than the available compute limit to cancel all orders

Reply:
Yes good idea, I will add a limit as an input to the function.